### PR TITLE
refactor: drop ember-functional-modifiers

### LINKED ISSUE #7: Refactor away from ember-functional-modifiers
https://github.com/jelhan/ember-style-modifier/blob/324445fc768b851f44578c675821356d8b3f92ba/addon/modifiers/style.js#L7-L11

This is a clear indicator that this modifier is in fact not _functional_ / a [pure function](https://en.wikipedia.org/wiki/Pure_function). 😄 

I would suggest to refactor this to not use `ember-functional-modifiers` (even though it's great!) and instead interact with the manager and state bucket directly. I'm doing this in [`ember-on-modifier`](https://github.com/buschtoens/ember-on-modifier/blob/700ce6dd79e5595410e5f397b4c2c95ddedb5727/addon/modifiers/on.js#L37-L82) for instance.

I'd be happy to submit a PR as well, if you like.

### PR DESCRIPTION
**Pull request by buschtoens**
This PR removes `ember-functional-modifiers` and instead directly interacts with the manager / state bucket.

Fixes #7.

### CODE REVIEW

**Comment by jelhan** [comment]
Thanks a lot!

I'm not 100%ly sure but I think there is an issue with removing of styles and normalization of property names. Property names stored in `state.styles` are not normalized. Therefor `oldStyles` passed to `setStyles` function may contain camelCase property names. But only dasherized spelling is removed from `rulesToRemove` array if property name exists in `newStyles`. I think we miss test coverage for this edge case but what I'm trying to describe is the following:

```js
await render(hbs`<p {{style styles}}></p>`);

this.set('styles', { fontSize: '16px' });
this.set('styles', { font-size: '16px' });
```

Maybe property names should be dasherized in `getStylesFromOptions` already? This must also consider edge cases like `{{style (hash fontSize="16px") font-size="8px"}}`.

File: addon/modifiers/style.js [hunk 1]
```diff
@@ -1,49 +1,70 @@
...
-// to clean up properties that were set but aren't
-// present in options hash anymore.
-const STORE = new WeakMap();
+const isObject = o => typeof o === 'object' && Boolean(o);
 
-export default makeFunctionalModifier((element, [optionsHash = {}], namedOptions = {}) => {
-  let options = assign({}, optionsHash, namedOptions);
-  let properties = [];
+const getStylesFromOptions = (positional, named) =>
+  assign({}, ...positional.filter(isObject), named);
```
Comment (jelhan) [inline addon/modifiers/style.js]: This adds support for passing multiple hashes to styles modifier isn't it? I like it. :+1: Maybe we should document it and add test coverage as well.

Comment (buschtoens) [reply addon/modifiers/style.js]: It does! :)

**Comment by buschtoens** [comment]
> Maybe property names should be dasherized in getStylesFromOptions already?

Good catch!

File: tests/integration/modifiers/style-test.js [hunk 2]
```diff
@@ -15,77 +15,108 @@ module('Integration | Modifiers | style', function(hooks) {
...
+
+      assert.dom('p').hasStyle({ fontSize: '12px', textAlign: 'right' });
+    });
+
+    test('supports multiple hashes', async function(assert) {
+      await render(
+        hbs`<p {{style (hash color="red" font-size="12px") (hash color="green")}}></p>`
+      );
+
+      assert.dom('p').hasStyle({ color: 'green', fontSize: '12px' });
```
Comment (jelhan) [inline tests/integration/modifiers/style-test.js]: `color` isn't the best property to test against:

```js
// Result:
{
  "color": "rgb(0, 128, 0)",
  "fontSize": "12px"
}
```

Replacing `color` with `display` should fix the tests.

**Comment by jelhan** [comment]
Thanks a lot!
